### ISSUE TITLE
h5py 3.0 will change default file open mode

### ISSUE THREAD

**Issue by randallpittman** [frejanordsiek/hdf5storage#82]
With hdf5storage 0.1.15 and h5py 2.10.0, I'm now getting the following deprecation warning:
```
/shared/.pyenv/versions/radar/lib/python3.8/site-packages/hdf5storage/__init__.py:1306: H5pyDeprecationWarning: The default file mode will change to 'r' (read-only) in h5py 3.0. To suppress this warning, pass the mode you need to h5py.File(), or set the global default h5.get_config().default_file_mode, or set the environment variable H5PY_DEFAULT_READONLY=1. Available modes are: 'r', 'r+', 'w', 'w-'/'x', 'a'. See the docs for details.
  f = h5py.File(filename)
```
Apparently any time the file mode is assumed to be "a" rather than "r" it needs to now be set to that explicitly.

**Comment by randallpittman** [comment]
From what I can tell of the current git source, that's the only h5py.File() call without mode set explicitly.

**Comment by randallpittman** [comment]
PR #81 addresses this.

**Comment by RizzerOnGitHub** [comment]
Also needs `mode='w'` in tests/test_hdf5_filters.py l.73

**Comment by frejanordsiek** [comment]
Fixed for the 0.1.x branch in commit 45af793

**Comment by frejanordsiek** [comment]
Fixed for master branch in 0e94e0f
